experimental/iterators/include/operators/column_scan.h

```diff
@@ -0,0 +1,50 @@
+#ifndef OPERATORS_COLUMN_SCAN_H
+#define OPERATORS_COLUMN_SCAN_H
+
+#include <optional>
+#include <tuple>
+#include <vector>
+
+template <typename... InputTypes>
+class ColumnScanOperator {
+ public:
+  using OutputTuple = std::tuple<InputTypes...>;
+  using ReturnType = std::optional<OutputTuple>;
+
+  explicit ColumnScanOperator(std::vector<InputTypes>... inputs)
+      : inputs_(std::make_tuple(std::move(inputs)...)), current_pos_(0) {}
+
+  void Open() {}
+  ReturnType ComputeNext() {
+    // Signal end-of-stream if we are at the end of the input
+    if (current_pos_ >= std::get<0>(inputs_).size()) {
+      return {};
+    }
+
+    // Return current tuple and advance
+    using IndexSequence =
+        std::make_index_sequence<std::tuple_size_v<OutputTuple>>;
+    auto const ret = ExtractCurrentTuple(IndexSequence{});
+    current_pos_++;
+    return ret;
+  }
+  void Close() {}
+
+ private:
+  template <std::size_t... kIndices>
+  std::optional<OutputTuple> ExtractCurrentTuple(
+      const std::index_sequence<kIndices...> & /*unused*/) const {
+    return std::make_tuple(std::get<kIndices>(inputs_).at(current_pos_)...);
+  }
+
+  std::tuple<std::vector<InputTypes>...> inputs_;
+  int64_t current_pos_;
+};
+
+template <typename... InputTypes>
+auto MakeColumnScanOperator(std::vector<InputTypes>... inputs) {
```

Comment:
Could you add some comments to explain the expected model of computation?

I am inferring that you want a "struct of array" input (or a tuple<vector<Type>>) and you are going to coiterate on all these and extract a tuple<Type>.

Using array of struct / struct of array terminology (in comments, in addition to DB terminology) would be nice to help brige the naming gap.

Reply:
I have added documentation with the latest commit. Let me know if you spot anything I should add.